### PR TITLE
[WIP] Implement opensearch engine

### PR DESCRIPTION
**Pull request by Felixoid**
I would like to have a big green button `add search engine` on my browser. There few things wich better to discuss, @hasegeli

### CODE REVIEW

File: serveradmin/common/templates/opensearch.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<OpenSearchDescription xmlns="http://a9.com/-/spec/opensearch/1.1/">
+<ShortName>ServerAdmin</ShortName>
+<Description>Server shell</Description>
+<Image width="16" height="16" type="image/x-icon">/favicon.ico</Image>
+<Url type="text/html" method="GET" template="{{ SCHEMA_FQDN }}/servershell/?term={searchTerms}">
+</Url>
+<Url type="application/opensearchdescription+xml" rel="self" template="{{ SCHEMA_FQDN }}/{{ STATIC_URL }}/opensearch.xml">
```
Comment (Felixoid) [inline serveradmin/common/templates/opensearch.xml]: This is not under jinja processing, but it should be automated somehow
@hasegeli

File: serveradmin/common/templates/opensearch.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<OpenSearchDescription xmlns="http://a9.com/-/spec/opensearch/1.1/">
+<ShortName>ServerAdmin</ShortName>
+<Description>Server shell</Description>
+<Image width="16" height="16" type="image/x-icon">/favicon.ico</Image>
+<Url type="text/html" method="GET" template="{{ SCHEMA_FQDN }}/servershell/?term={searchTerms}">
```
Comment (Felixoid) [inline serveradmin/common/templates/opensearch.xml]: same here

File: serveradmin/common/templates/base.html [hunk 3]
```diff
@@ -15,6 +15,7 @@
 <script type="text/javascript" src="{{ STATIC_URL }}common.js"></script>
 {% block additional_scripts %}
 {% endblock %}
+<link rel="search" type="application/opensearchdescription+xml" href="{{ STATIC_URL }}opensearch.xml" />
```
Comment (Felixoid) [inline serveradmin/common/templates/base.html]: If the file doesn't exist nothing shouldn't break

Comment (hasegeli) [reply serveradmin/common/templates/base.html]: I don't understand.  Why the file wouldn't exist?  You place it there.

Comment (hasegeli) [inline serveradmin/common/templates/base.html]: Templates are not under `STATIC_URL`.

File: serveradmin/common/templates/opensearch.xml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<OpenSearchDescription xmlns="http://a9.com/-/spec/opensearch/1.1/">
+<ShortName>ServerAdmin</ShortName>
```
Comment (hasegeli) [inline serveradmin/common/templates/opensearch.xml]: Serveradmin is written together.  Please do not capitalize `A`.

File: serveradmin/common/templates/opensearch.xml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+<Description>Server shell</Description>
```
Comment (hasegeli) [inline serveradmin/common/templates/opensearch.xml]: Servershell is also written together.  `Description` should probably say something else anyway.

File: serveradmin/common/templates/opensearch.xml (hunk 2 above)
Comment (hasegeli) [inline serveradmin/common/templates/opensearch.xml]: What is `SCHEMA_FQDN`?

File: serveradmin/common/templates/opensearch.xml [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+<Image width="16" height="16" type="image/x-icon">/favicon.ico</Image>
+<Url type="text/html" method="GET" template="{{ SCHEMA_FQDN }}/servershell/?term={searchTerms}">
+</Url>
+<Url type="application/opensearchdescription+xml" rel="self" template="{{ SCHEMA_FQDN }}/{{ STATIC_URL }}/opensearch.xml">
+</Url>
+<InputEncoding>UTF-8</InputEncoding>
+</OpenSearchDescription>
```
Comment (hasegeli) [inline serveradmin/common/templates/opensearch.xml]: You need to a view to use this template.  I think `servershell` would be the proper module for it.  See `urls.py` and `views.py`.  I don't think you can make this work without testing your changes.

File: serveradmin/common/templates/opensearch.xml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<OpenSearchDescription xmlns="http://a9.com/-/spec/opensearch/1.1/">
+<ShortName>ServerAdmin</ShortName>
+<Description>Server shell</Description>
+<Image width="16" height="16" type="image/x-icon">/favicon.ico</Image>
+<Url type="text/html" method="GET" template="{{ SCHEMA_FQDN }}/servershell/?term={searchTerms}">
+</Url>
+<Url type="application/opensearchdescription+xml" rel="self" template="{{ SCHEMA_FQDN }}/{{ STATIC_URL }}/opensearch.xml">
+</Url>
+<InputEncoding>UTF-8</InputEncoding>
```
Comment (hasegeli) [inline serveradmin/common/templates/opensearch.xml]: I don't think we actually support `UTF-8`.

File: serveradmin/common/templates/opensearch.xml (hunk 1 above)
Comment (hasegeli) [reply serveradmin/common/templates/opensearch.xml]: I couldn't understand.

File: serveradmin/common/templates/opensearch.xml (hunk 2 above)
Comment (Felixoid) [reply serveradmin/common/templates/opensearch.xml]: this is really draft, I would like to discuss how to do it properly

**Pull request closed without merging**